client/admin/sidebar/AdminSidebar.js

```diff
@@ -0,0 +1,128 @@
+import React, { useCallback, useState, useMemo } from 'react';
+import { Box, Button, Icon, SearchInput, Scrollable, Skeleton } from '@rocket.chat/fuselage';
+import { css } from '@rocket.chat/css-in-js';
+
+import { menu, SideNav, Layout } from '../../../app/ui-utils/client';
+import { useReactiveValue } from '../../hooks/useReactiveValue';
+import { useTranslation } from '../../contexts/TranslationContext';
+import { useRoutePath, useCurrentRoute } from '../../contexts/RouterContext';
+import { useAtLeastOnePermission } from '../../contexts/AuthorizationContext';
+import { sidebarItems } from '../sidebarItems';
+import { useSettingsGroupsFiltered } from './useSettingsGroupsFiltered';
+
+const SidebarItem = ({ permissionGranted, pathGroup, href, icon, label, currentPath }) => {
+	if (permissionGranted && !permissionGranted()) { return null; }
+	const params = useMemo(() => ({ group: pathGroup }), [pathGroup]);
+	const path = useRoutePath(href, params);
+	const isActive = path === currentPath || false;
+	return useMemo(() => <Box
```

Comment:
why not React.memo?

Reply:
@ggazzo `React.memo()` returns a new component, which should be declared outside this component scope.

Reply:
sure I know , but why not wrap the component as a memoized component instead if this approach?